resources/providers/azurelib/provider.go

```diff
@@ -57,52 +59,56 @@ func (p *ProviderInitializer) Init(log *logp.Logger, azureConfig auth.AzureFacto
 		return nil, fmt.Errorf("failed to init monitor client: %w", err)
 	}
 
-	inventoryProvider := inventory.NewProvider(log, resourceGraphClientFactory, diagnosticSettingsClient, azureConfig.Credentials)
+	inventoryProvider := inventory.NewAssetsARGProvider(log, resourceGraphClientFactory)
 	return &provider{
-		inventory:  inventoryProvider,
-		governance: governance.NewProvider(log, inventoryProvider),
+		inventory:        inventoryProvider,
+		sqlInventory:     inventory.NewSQLProvider(log, azureConfig.Credentials),
+		storageInventory: inventory.NewStorageAccountProvider(log, diagnosticSettingsClient, azureConfig.Credentials),
+		governance:       governance.NewProvider(log, inventoryProvider),
 	}, nil
 }
 
 type provider struct {
-	inventory  inventory.ProviderAPI
-	governance governance.ProviderAPI
+	inventory        inventory.AssetsARGProviderAPI
+	sqlInventory     inventory.SQLProviderAPI
+	storageInventory inventory.StorageAccountProviderAPI
+	governance       governance.ProviderAPI
```

Comment:
[Suggestion]
Since we are refactoring, perhaps we could just ebbed the interfaces to avoid rewriting the function signatures for each function of each interface. 

```golang
type provider struct {
	inventory.AssetsARGProviderAPI
	inventory.SQLProviderAPI
	inventory.StorageAccountProviderAPI
	governance.ProviderAPI
}
```

Reply:
Done!